src/serialize.js

```diff
@@ -20,7 +20,7 @@ const toProtoBuf = (node) => {
   if (node.Links && node.Links.length > 0) {
     pbn.Links = node.Links
       .map((link) => ({
-        Hash: link.Hash.buffer,
+        Hash: asBuffer(link.Hash.buffer),
```

Comment:
Did you mean to pass `link.Hash` in here instead?

If `link.Hash` is a node buffer, the `.buffer` property is an ArrayBuffer that might not correspond exactly to the original bytes.

We need to use `Buffer.from`, `link.Hash.byteOffset` and `link.Hash.length` to make sure we don't end up with the wrong bytes if we're accessing the backing ArrayBuffer directly - https://nodejs.org/api/buffer.html#buffer_buf_buffer

As it stands it'll get turned back into a node Buffer further down at line 44 that contains the whole ArrayBuffer.

Reply:
Good catch, I meant `link.Hash` indeed (I think I've overlooked `link.Hash` was Buffer rather than thing that had `{buffer: Buffer}`).

_Fact that no test caught this makes me quite a bit nervous._

Reply:
Wait no, `link.Hash` is a `CID` not the `Buffer`

Reply:
New commit removes this changes in favor of adding Uint8Array support into protons library instead (See https://github.com/ipfs/protons/pull/13)

Reply:
> Wait no, link.Hash is a CID not the Buffer

Quite right, this is why I shouldn't look at these things in the evening!